src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/CVService/CVObjectDetectionRequest.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Text;
+using System.Threading;
+using JetBrains.Annotations;
+using UnityEngine;
+using RegressionGames.StateRecorder.JsonConverters;
+
+namespace RegressionGames.StateRecorder.BotSegments.Models.CVService
+{
+    public class CVObjectDetectionRequest
+    {
+        public CVImageBinaryData screenshot;
+        [CanBeNull] public CVImageBinaryData queryImage;
+        [CanBeNull] public string queryText;
+        /*
+         * <summary>Rect relative to the screenshot resolution</summary>
+        */
+        public RectInt? withinRect;
+
+        // track the index in this bot segment for correlation of the responses
+        public int index;
+
+        
+        private static readonly ThreadLocal<StringBuilder> _stringBuilder = new (() => new(1000));
+        public string ToJsonString()
+        {
+            _stringBuilder.Value.Clear();
+            WriteToStringBuilder(_stringBuilder.Value);
+            return _stringBuilder.Value.ToString();
+        }
+
+        public void WriteToStringBuilder(StringBuilder stringBuilder)
+        {
+
+            bool textQuery = queryText != null;
+            bool imageQuery = queryImage != null;
+
+            if (textQuery && imageQuery)
+            {
+                RGDebug.LogError("You need to use either textQuery or imageQuery. You can not have both.");
```

Comment:
Does our WriteToStringBuilder approach throw errors in other instances? Trying to determine if this should be caught here or elsewhere.

Reply:
No, you are definitely right. This is the wrong place to do it. I think we should do it in the constructor

Reply:
I thought about this a little more. This is better but not good enough. Because it is likely the constructor will never actually receive both of these values, Especially if they are set as they are now, like so:

```
                    _ = CVServiceManager.GetInstance().PostCriteriaObjectTextQuery(
                        new CVObjectDetectionRequest(
                            screenshot: new CVImageBinaryData()
                            {
                                width = width,
                                height = height,
                                data = imageData
                            },
                            queryText: queryText,
                            queryImage: null,
                            withinRect: null,
                            index: 1
                        ),
```

Where this check needs to live is in parsing from the json. If in the criteria match json they give both text AND image they should be notified then and there. I added a TODO and the corresponding task.

Reply:
Ended up fixing it in this PR. However, I have a question about logging errors:
![image](https://github.com/user-attachments/assets/9e004e29-9771-4350-afd5-143983ec3cf9)

With both 
```
RGDebug.LogError("Both textQuery and imageQuery cannot be provided simultaneously. Use only one.");
throw new ArgumentException("Both textQuery and imageQuery cannot be provided simultaneously. Use only one.");
```
I feel like we are duplicating information? Should RGDebug.Log actually raise an error?

Reply:
yeah I'm looking through our code and I am confused about how exactly the pattern would work, I can't find any spot where we throw an error... it seems like we always just terminate, stop loops, etc... 

I suppose putting it into that CVObjectDetectionKeyFrameCriteriaData parsing file is fine... there is an argument to be made that this is not an error for the JSON parsing and is more of an error for the business logic later on... but it gets weird since we are defining this data format and parsing the JSON in the same class, so the class could impose the restrictions I suppose... this seems fine to me!

TLDR; yeah I think this is all good for now, but let's make a note for @RG-nAmKcAz to check this out when he gets back in case he has a pattern in mind that we were following.

Reply:
(mark this as resolved when you read it)